app/questionnaire/path_finder.py

```diff
@@ -198,8 +200,49 @@ def get_next_location(self, current_location):
         current_location_index = PathFinder._get_current_location_index(routing_path, current_location)
 
         if current_location_index is not None and current_location_index < len(routing_path) - 1:
+            completed_blocks = get_completed_blocks(current_user)
+            if self._is_survey_completed(routing_path, completed_blocks):
+                return routing_path[-1]     # Go to Summary location
+
+            section_summary_location = self._get_valid_section_summary(current_location.block_id, routing_path, completed_blocks)
+            if section_summary_location:
+                return section_summary_location     # Go to SectionSummary location
+
             return routing_path[current_location_index + 1]
 
+    @staticmethod
+    def _is_survey_completed(routing_path, completed_blocks):
+        # Check all blocks on routing path are complete
+        for location in routing_path[:-1]:  # Don't evaluate end block (Summary or Confirmation)
```

Comment:
We are assuming here that all schemas will either finish with summary or conformation?

Reply:
For every schema currently in the repo this is true. I think we depend on this being the case in other places as well as this.

Let's check that validator enforces this.

Reply:
I just checked - [it does](https://github.com/ONSdigital/eq-schema-validator/blob/master/app/validation/validator.py#L212).

Reply:
Makes sense just wanted to check that we should enforce this

Reply:
Yes the last block should be a summary or confirmation page. Regardless the last block will never be completed at this point otherwise the questionnaire would already have been submitted.